LatentGoalDnf[] list_latent
int32 max_x
int32 max_y